caluma/caluma_form/tests/test_document.py

```diff
@@ -511,269 +509,35 @@ def test_save_document(
-            None,
-            "SaveDocumentStringAnswer",
-            False,
-        ),
-        (
-            Question.TYPE_TEXTAREA,
-            {},
-            None,
-            ["email"],
-            "test@example.com",
-            None,
-            "SaveDocumentStringAnswer",
-            True,
-        ),
+        (Question.TYPE_FLOAT, {"min_value": 100.0}, None, [], 1, None, "SaveDocumentFloatAnswer", False),
+        (Question.TYPE_TEXT, {}, None, [], "Test", None, "SaveDocumentStringAnswer", True),
+        (Question.TYPE_TEXT, {"max_length": 1}, None, [], "toolong", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_DATE, {}, None, [], None, "1900-01-01", "SaveDocumentDateAnswer", False),
+        (Question.TYPE_DATE, {}, None, [], None, "2019-02-22", "SaveDocumentDateAnswer", True),
+        (Question.TYPE_FILES, {}, None, [], None, None, "SaveDocumentFilesAnswer", False),
+        (Question.TYPE_FILES, {}, None, [], [{"name": "some-file.pdf"}], None, "SaveDocumentFilesAnswer", True),
+        (Question.TYPE_FILES, {}, None, [], [{"name": "not-exist.pdf"}], None, "SaveDocumentFilesAnswer", True),
+        (Question.TYPE_TEXT, {"min_length": 10}, None, [], "tooshort", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_TABLE, {}, None, [], None, None, "SaveDocumentTableAnswer", True),
+        (Question.TYPE_TEXTAREA, {}, None, [], "Test", None, "SaveDocumentStringAnswer", True),
+        (Question.TYPE_TEXTAREA, {"max_length": 1}, None, [], "toolong", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_MULTIPLE_CHOICE, {}, None, [], ["option-slug"], None, "SaveDocumentListAnswer", True),
+        (Question.TYPE_MULTIPLE_CHOICE, {}, None, [], ["option-slug", "option-invalid-slug"], None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_CHOICE, {}, None, [], "option-slug", None, "SaveDocumentStringAnswer", True),
+        (Question.TYPE_CHOICE, {}, None, [], "invalid-option-slug", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_DYNAMIC_MULTIPLE_CHOICE, {}, "MyDataSource", [], ["5.5", "1"], None, "SaveDocumentListAnswer", True),
+        (Question.TYPE_DYNAMIC_MULTIPLE_CHOICE, {}, "MyDataSource", [], ["not in data"], None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_DYNAMIC_CHOICE, {}, "MyDataSource", [], "5.5", None, "SaveDocumentStringAnswer", True),
+        (Question.TYPE_DYNAMIC_CHOICE, {}, "MyDataSource", [], "not in data", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_TEXT, {}, None, ["email"], "some text", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_TEXT, {}, None, ["email"], "test@example.com", None, "SaveDocumentStringAnswer", True),
+        (Question.TYPE_TEXTAREA, {}, None, ["email"], "some text", None, "SaveDocumentStringAnswer", False),
+        (Question.TYPE_TEXTAREA, {}, None, ["email"], "test@example.com", None, "SaveDocumentStringAnswer", True),
     ],
-)
+)  # fmt:skip
```

Comment:
drive-by fix: exclude this block from autoformatting so that all parametrizations can stay one one line